### PR TITLE
[Merged by Bors] - chore: split `RingTheory.Ideal.Operations`

### PR DESCRIPTION
**Pull request by xroblot**
Split the file `RingTheory.Ideal.Operations` into two parts by putting everything related to maps into a new file called... `RingTheory.Ideal.Maps`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by adomani** [comment]
In case it is useful, the `no_lost_declarations` scripts confirms that nothing was lost!

---

|File|+-|Declaration|
|-|-|-|
---
* 0  added declarations
* 0  removed declarations
* 113  paired declarations
---

You can run this locally using
```bash
./scripts/no_lost_declarations.sh <optional_commit>
```

**Comment by xroblot** [comment]
> In case it is useful, the `no_lost_declarations` scripts confirms that nothing was lost!
> 
> File	+-	Declaration
> * 0  added declarations
> * 0  removed declarations
> * 113  paired declarations
> 
> You can run this locally using
> 
> ```shell
> ./scripts/no_lost_declarations.sh <optional_commit>
> ```

Thanks. That's very helpful!

**Comment by xroblot** [comment]
Wait. I don't see the file `no_lost_declarations.sh` in the `scripts` directory? Am I missing something?

**Comment by adomani** [comment]
Ah, the script is in a PR: I simply copy-pasted the output of the script here.

**Comment by xroblot** [comment]
I see. I hope it gets merged soon since it appears to be very helpful.

File: Mathlib/Algebra/CharP/Quotient.lean [hunk 1]
```diff
@@ -4,8 +4,9 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Kenny Lau, Eric Wieser
 -/
 import Mathlib.Algebra.CharP.Basic
+import Mathlib.RingTheory.Ideal.Maps
 import Mathlib.RingTheory.Ideal.Quotient
-import Mathlib.RingTheory.Ideal.Operations
+
```
Comment (YaelDillies) [inline Mathlib/Algebra/CharP/Quotient.lean:9]:
```suggestion
```

File: Mathlib/RingTheory/Ideal/Maps.lean [hunk 2]
```diff
@@ -0,0 +1,913 @@
...
+  add_mem' {x y} hx hy := by
+    simp only [Set.mem_preimage, SetLike.mem_coe, map_add f] at hx hy ⊢
+    exact add_mem hx hy
+  zero_mem' := by simp only [Set.mem_preimage, map_zero, SetLike.mem_coe, Submodule.zero_mem]
+  smul_mem' c x hx := by
+    simp only [smul_eq_mul, Set.mem_preimage, map_mul, SetLike.mem_coe] at *
+    exact mul_mem_left I _ hx
+#align ideal.comap Ideal.comap
+
+-- Porting note (#10756): new theorem
```
Comment (YaelDillies) [inline Mathlib/RingTheory/Ideal/Maps.lean:52]: I think you might as well drop that porting note

File: Mathlib/RingTheory/Ideal/Operations.lean
Comment (YaelDillies) [inline Mathlib/RingTheory/Ideal/Operations.lean:1]: Would be nice to expand the module docstring to say what's actually in that file.

**Review by YaelDillies — COMMENTED** [review]
This split looks very reasonable.

File: Mathlib/RingTheory/Ideal/Operations.lean
Comment (xroblot) [reply Mathlib/RingTheory/Ideal/Operations.lean:1]: Sure. But I guess that'll be for another PR 😅

File: Mathlib/RingTheory/Ideal/Maps.lean (hunk 2 above)
Comment (xroblot) [reply Mathlib/RingTheory/Ideal/Maps.lean]: I also addressed the note on the next line
```lean
-- @[simp] -- Porting note (#11215): TODO enable simp after the port
```
and enabled the simp. Let's see if that works...

Comment (xroblot) [reply Mathlib/RingTheory/Ideal/Maps.lean]: Looks fine.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**